fxtest/lifecycle.go

```diff
@@ -116,7 +133,23 @@ func (l *Lifecycle) RequireStart() *Lifecycle {
 // If any hook returns an error, execution continues for a best-effort
 // cleanup. Any errors encountered are collected into a single error and
 // returned.
-func (l *Lifecycle) Stop(ctx context.Context) error { return l.lc.Stop(ctx) }
+func (l *Lifecycle) Stop(ctx context.Context) error {
+	c := make(chan error, 1)
+	callerChan := make(chan string, 1)
+	recordChan := make(chan lifecycle.HookRecord, 1)
+
+	go func() { c <- l.lc.Stop(ctx, callerChan, recordChan) }()
+	for {
+		select {
+		case err := <-c:
+			return err
+		case <-callerChan:
+			continue
+		case <-recordChan:
+			continue
+		}
+	}
+}
```

Comment:
eh, this is identical (other than `lc` component) to above and an extra `// Ignore caller/hookrecord channel`. i dunno if its worth refactoring a bit for easier testing?